fastlane_core/lib/fastlane_core/configuration/configuration.rb

```diff
@@ -243,11 +243,17 @@ def fetch(key, ask: true)
       while value.nil?
         UI.important("To not be asked about this value, you can specify it using '#{option.key}'") if ENV["FASTLANE_ONBOARDING_IN_PROCESS"].to_s.length == 0
         value = option.sensitive ? UI.password("#{option.description}: ") : UI.input("#{option.description}: ")
+
+        # ConfigItem allows to specify a type for the item but UI.password and
+        # UI.input return String values. Try to convert the String input to
+        # the option's type before passing it along.
+        value = option.auto_convert_value(value)
+
         # Also store this value to use it from now on
         begin
           set(key, value)
         rescue => ex
-          puts(ex)
+          UI.error(ex)
```

Comment:
I didn't go for `user_error!` because it would make Fastlane exit, right? Whereas here we just want to show an error and give the user another chance to insert the correct value.